third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -182,4 +184,14 @@ impl<'a, 'b, 'c> NativeContext<'a, 'b, 'c> {
     pub fn gas_balance(&self) -> InternalGas {
         self.gas_balance
     }
+
+    pub fn load_function(
+        &mut self,
+        module: &ModuleId,
+        function_name: &Identifier,
+    ) -> PartialVMResult<Arc<Function>> {
+        self.resolver
+            .module_store()
+            .resolve_function_by_name(function_name, module)
```

Comment:
@runtian-zhou I think in order to prevent this function from being accidentally misused and result in non-determinism, we should check if the module has already been marked as visited in the traversal context.

We should be able to pass a reference to the traversal context to the native context so it can be accessed here.